[BUG]Compilation errors

**Describe the bug**
Some issues with spritesheets are causing compilation errors.  The errors I am getting are:
`Assets\Scripts\SpriteAnimator.cs(27,13): error CS0019: Operator '==' cannot be applied to operands of type 'SpriteSheet' and 'SpriteSheet'`
`Assets\Scripts\BlockSystem\BuildSystem.cs(143,76): error CS0037: Cannot convert null to 'SpriteSheet' because it is a non-nullable value type`

**To Reproduce**
Steps to reproduce the behavior:
1. Attempt to compile the game

**Expected behavior**
The game compiles

**Screenshots**
If applicable, add screenshots to help explain your problem.

**System Information (please complete the following information):**
 - OS: Windows
 - Game version: Latest dev branch

**Additional context**

--- a/Blocktest/Assets/Scripts/BlockSystem/Block.cs
+++ b/Blocktest/Assets/Scripts/BlockSystem/Block.cs
@@ -23,7 +23,7 @@ namespace BlockSystem
         /// <summary> The sound that is played when the block is placed. </summary>
         public AudioClip placeSound;
         /// <summary> The sprite sheet used for smoothing the block. </summary>
-        public SpriteSheet spriteSheet;
+        public SpriteSheet? spriteSheet;
 
         /* METHODS */
 
@@ -42,7 +42,7 @@ namespace BlockSystem
             }
             if (!blockSmoothing) { return; }
             spriteSheet = new SpriteSheet(path);
-            if (spriteSheet.Length <= 1) {
+            if (spriteSheet?.Length <= 1) {
                 Debug.Log("Block " + this + " is marked as smoothable, but a sprite sheet could not be found at " + path + "!");
             }
         }

--- a/Blocktest/Assets/Scripts/BlockSystem/BuildSystem.cs
+++ b/Blocktest/Assets/Scripts/BlockSystem/BuildSystem.cs
@@ -140,7 +140,7 @@ namespace BlockSystem
 
         public override void GetTileData(Vector3Int position, ITilemap tilemap, ref TileData tileData)
         {
-            if (!sourceBlock.blockSmoothing || (sourceBlock.spriteSheet is null)) {
+            if (!sourceBlock.blockSmoothing /*|| (sourceBlock.spriteSheet is null)*/) {
                 base.GetTileData(position, tilemap, ref tileData);
                 return;
             } // If the tile doesn't or can't smooth, don't even try
@@ -160,7 +160,7 @@ namespace BlockSystem
                 bitmask += 8;
             }
 
-            sprite = sourceBlock.spriteSheet[sourceBlock.blockSprite.texture.name + "_" + bitmask];
+            sprite = sourceBlock.spriteSheet?[sourceBlock.blockSprite.texture.name + "_" + bitmask];
             base.GetTileData(position, tilemap, ref tileData);
         }
 

--- a/Blocktest/Assets/Scripts/BlockSystem/BuildSystem.cs
+++ b/Blocktest/Assets/Scripts/BlockSystem/BuildSystem.cs
@@ -140,7 +140,7 @@ namespace BlockSystem
 
         public override void GetTileData(Vector3Int position, ITilemap tilemap, ref TileData tileData)
         {
-            if (!sourceBlock.blockSmoothing /*|| (sourceBlock.spriteSheet is null)*/) {
+            if (!sourceBlock.blockSmoothing || (sourceBlock.spriteSheet is null)) {
                 base.GetTileData(position, tilemap, ref tileData);
                 return;
             } // If the tile doesn't or can't smooth, don't even try

--- a/Blocktest/Assets/Scripts/SpriteAnimator.cs
+++ b/Blocktest/Assets/Scripts/SpriteAnimator.cs
@@ -24,7 +24,7 @@ public class SpriteAnimator : MonoBehaviour
 
     public void SetAnimation(SpriteSheet newSheet, float newTime = 0.1f)
     {
-        if (newSheet == spriteSheet) { return; }
+        if (newSheet.Equals(spriteSheet)) { return; }
         spriteRenderer.sprite = newSheet[0];
         currentSprite = 0;
         spriteSheet = newSheet;

